packages/playwright/src/worker/testInfo.ts

```diff
@@ -448,8 +448,12 @@ export class TestInfoImpl implements TestInfo {
     return sanitizeForFilePath(trimLongString(fullTitleWithoutSpec));
   }
 
-  _resolveSnapshotPath(template: string | undefined, defaultTemplate: string, pathSegments: string[], extension?: string) {
-    const subPath = path.join(...pathSegments);
+  _resolveSnapshotPath(template: string | undefined, defaultTemplate: string, pathSegments: string[], sanitizeFilePath: boolean, extension?: string) {
+    let segments = pathSegments;
+    if (sanitizeFilePath)
+      segments = pathSegments.map((segment, index) => sanitizeFilePathBeforeExtension(segment, index === pathSegments.length - 1 ? extension : undefined));
```

Comment:
why not join first and sanitize after? you'd not need the conditional extension.